[Site] Clean cook

smnandre opened the pull request.

smnandre committed:
[Site] Clean cook

Kocal commented on ux.symfony.com/assets/images/cookbook/component-architecture-1280x720.png:
Due to the number of colors, I don't think PNG is a good format for this image.
With a dump PNG to JPEG conversion (https://png2jpg.com/), the same image in JPEG format is about ~220ko without too much loss.

Also, what about modern formats like AVIF or WEBP?

smnandre replied:
I reduced and compressed the existing one ;) 

Agree with you here, we'll talk about it later in a more "macro" scope if possible.

Kocal commented on ux.symfony.com/assets/images/cookbook/component-architecture-640x360.png:
Same comment here

Kocal commented on ux.symfony.com/cookbook/component_architecture.md:
```diff
@@ -174,5 +174,5 @@ class Button
-Follow those rules help your front developpers working on codebase
-their are familiar with since those rules are already used in the js world.
+Follow those rules help your front developers working on codebase
+they are familiar with since those rules are already used in the JS world.
```
`JS` or `JavaScript`?

smnandre replied:
Quick fixes as i saw them.

 Could you open a PR/issue if you review the content (grammar / typos / style) ?

Kocal commented on ux.symfony.com/src/Service/CookbookFactory.php:
```diff
@@ -12,27 +12,65 @@
+            throw new \RuntimeException(\sprintf('Cookbook file "%s" does not contains Front Matter data.', $file));
+        }
+        if (!isset($frontMatter['title']) || !\is_string($frontMatter['title'])) {
+            throw new \RuntimeException('Missing title in Front Matter.');
```
```suggestion
            throw new \RuntimeException('Front Matter data "title" is missing, or is not a string.');
```

Kocal replied:
Same comments for the description/image/tags

smnandre replied:
Should be a logical exception here to me, with no comments. I will refactor this later, so here is fine to me for now.

Kocal commented on ux.symfony.com/src/Service/CookbookFactory.php:
```diff
@@ -12,27 +12,65 @@
+
+    private function getConverter(): MarkdownConverter
+    {
+        if (isset($this->converter)) {
+            return $this->converter;
+        }
+
+        $environment = new Environment();
+        $environment->addExtension(new CommonMarkCoreExtension());
+        $environment->addExtension(new FrontMatterExtension());
+
+        return $this->converter = new MarkdownConverter($environment);
```
Very minor, but I find this more easily understandable with this way:
```suggestion
        if (!isset($this->converter)) {
            $environment = new Environment();
            $environment->addExtension(new CommonMarkCoreExtension());
            $environment->addExtension(new FrontMatterExtension());

            $this->converter = new MarkdownConverter($environment)
        }

        return $this->converter;
```

Kocal commented on ux.symfony.com/templates/cookbook/index.html.twig:
```diff
@@ -4,28 +4,38 @@
+                    image="{{ asset('images/cookbook/%s-1280x720.png'|format(cookbook.slug)) }}"
+                    url="{{ path('app_cookbook_show', {slug: cookbook.slug}) }}"
+                    tags="{{ cookbook.tags }}"
+                    lazyload="{{ loop.index > 3 }}"
```
Since you lazyload images after the 3rd card, what about preloading and configure the `fetchpriority` to `high` for images from the 3 first cards?

smnandre replied:
No need to and would require more code. I just don't want to add a lazy for something in FCP

Kocal reviewed (commented):
Some comments

WebMamba commented:
I ran the PR locally and it looks like the code is not highlighted anymore:

<img width="1133" alt="Capture d’écran 2024-08-01 à 14 22 07" src="https://github.com/user-attachments/assets/e5347077-b9ec-4f40-a818-42d45374226d">

smnandre commented:
It was ? I'll look at it

smnandre commented:
Did you check when you rebased last time ? 

What I suppose here is that when you coded it, we used highlight js (hence the "`<code language-php>`".

Since we switched to tempest php. And it requires some tags around in Twig.

I'll look at it

smnandre commented:
Well i've done it for you @WebMamba  ❤️ 

It's even simpler in the end, i removed all the markdown parsing/formatting from the Factory (except the FrontMatter but this will be cached soon).

The content is rendered like a template with the current Twig environment + extensions. 

(and that allow us to _not_ parse the markdown content when we don't display it, like on the cookbook index page)

Also fixed the CSS (code was not readable on light mode)

kbond commented:
Thanks Simon.
